core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/IngestUtils.java

```diff
@@ -62,23 +84,213 @@
 					LOGGER.warn("Adapter not found for " + geowaveData.getValue());
 					continue;
 				}
-				final IndexWriter indexWriter;
-				if (idx.getId().equals(
-						geowaveData.getIndexId())) {
-					indexWriter = primaryIndexWriter;
-				}
-				else {
-					final PrimaryIndex index = requiredIndexMap.get(geowaveData.getIndexId());
-					if (index == null) {
-						LOGGER.warn("Index '" + geowaveData.getIndexId().getString() + "' not found for " + geowaveData.getValue());
-						continue;
+				IndexWriter indexWriter;
+				for (final ByteArrayId indexId : geowaveData.getIndexIds()) {
+					indexWriter = dimensionalityIndexMap.get(indexId);
+					if (indexWriter == null) {
+						final PrimaryIndex index = requiredIndexMap.get(indexId);
+						if (index == null) {
+							LOGGER.warn("Index '" + indexId.getString() + "' not found for " + geowaveData.getValue());
+							continue;
+						}
+						indexWriter = ingestRunData.getIndexWriter(index);
 					}
-					indexWriter = ingestRunData.getIndexWriter(index);
+					indexWriter.write(
+							adapter,
+							geowaveData.getValue());
+				}
+			}
+		}
+	}
+
+	public static PrimaryIndex getIndex(
```

Comment:
Was thinking that we could have an option to select an existing index.
Is it easier to include a custom provider for specific index or is it easier to pre-create a custom index via some index creation tool with many options per index type.  Perhaps we create a ticket for this later option since it seems like it is a bit much for the scope of this ticket.